packages/plugin/src/Bundles/Fields/Validation/CalculationFieldValidation.php

```diff
@@ -64,7 +64,15 @@ public function validate(ValidateEvent $event): void
             }
 
             if ($valueOrdination != $result) {
-                $field->addError(Freeform::t('Incorrectly calculated value'));
+                $errorMessage = Freeform::t('Incorrectly calculated value');
+
+                if (!$canRender) {
+                    $form->addError($errorMessage);
+
+                    return;
+                }
+
+                $field->addError($errorMessage);
```

Comment:
If possible, lets strive to use positive evaluations first, they're a bit easier to read when going through the code fast.



```suggestion
                $errorMessage = Freeform::t('Incorrectly calculated value');
                if ($canRender) {
                    $field->addError($errorMessage);
                } else {
                    $form->addError($errorMessage);
                }
```